### ISSUE TITLE
pessimistic version constraint

### ISSUE THREAD

**Issue by srhopkins** [petems/vagrant-puppet-install#26]
I was trying to use this functionality to install '~> 3.7' without success and also tried with '~> 2.7' per your spec config_spec.rb.

```
$ cat Vagrantfile
Vagrant.configure("2") do |config|
  config.vm.box = "puppetlabs/centos-7.0-64-nocm"

  # REQUIRES # vagrant plugin install vagrant-puppet-install
  config.puppet_install.puppet_version = "~> 2.7"

  # http://docs.vagrantup.com/v2/provisioning/puppet_apply.html
  config.vm.provision "puppet" do |puppet|
    puppet.options = "--verbose --debug"

    puppet.manifests_path = "."
    puppet.manifest_file = "default.pp"

    # ENVIRONMENTS
    #puppet.environment_path = "../../puppet/environments/"
    puppet.environment = "production"

    # MODULES
    puppet.module_path = "../../puppet/modules"

  end
end
```
... RESULTS IN ...

```
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

sh install.sh -v \~\>\ 2.7 2>&1

Stdout from the command:

00:11:41 +0000 INFO: Version parameter defined: ~> 2.7
00:11:41 +0000 INFO: Downloading Puppet ~> 2.7 for el...
00:11:41 +0000 INFO: Red hat like platform! Lets get you an RPM...
00:11:41 +0000 INFO: Downloading http://yum.puppetlabs.com/el/7/products/x86_64/puppetlabs-release-7-11.noarch.rpm
00:11:41 +0000 INFO:   to file /tmp/install.sh.3439.34247/puppetlabs-release-7-11.noarch.rpm
00:11:41 +0000 INFO: Trying wget...
00:11:41 +0000 INFO: installing puppetlabs yum repo with rpm...
warning: /tmp/install.sh.3439.34247/puppetlabs-release-7-11.noarch.rpm: Header V4 RSA/SHA1 Signature, key ID 4bd6ec30: NOKEY
Preparing...                          ########################################
Updating / installing...
puppetlabs-release-7-11               ########################################
Loaded plugins: fastestmirror
Determining fastest mirrors
 * base: centos-distro.cavecreek.net
 * extras: mirror.oss.ou.edu
 * updates: mirrors.psychz.net
No package puppet-~> 2.7 available.
Error: Nothing to do
00:12:00 +0000 CRIT: Installation failed
00:12:00 +0000 CRIT: Please file a bug report at https://github.com/petems/puppet-install-shell/
00:12:00 +0000 CRIT:
00:12:00 +0000 CRIT: Version: ~> 2.7
00:12:00 +0000 CRIT: Platform: el
00:12:00 +0000 CRIT: Platform Version: 7
00:12:00 +0000 CRIT: Machine: x86_64
00:12:00 +0000 CRIT: OS: Linux
00:12:00 +0000 CRIT:
00:12:00 +0000 CRIT: Please detail your operating system type, version and any other relevant details


Stderr from the command: 
```

**Comment by srhopkins** [comment]
Submitted #27

**Comment by srhopkins** [comment]
Fixed in #28. 

Thanks for developing this code; very useful and much appreciated!
